source/agora/common/Hash.d

```diff
@@ -119,15 +127,13 @@ public void hashPart (T) (scope const auto ref T record, scope HashDg state)
     /*pure*/ nothrow @nogc
     if (is(T == struct))
 {
-    static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
+    static if(is(typeof(T.init.computeHash(HashDg.init))))
+        record.computeHash(state);
+    else static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
         state(record[]);
     else
-    {
-        static assert(is(typeof(T.init.computeHash(HashDg.init))),
-                      "Struct `" ~ T.stringof ~
-                      "` does not implement `computeHash(scope HashDg) const nothrow @nogc` function");
-        record.computeHash(state);
-    }
+        foreach (ref field; record.tupleof)
```

Comment:
`const ref` or?

Reply:
Are all "tuple of type" except in case of Includes computeHash or "ubyte[]"?

Reply:
I'm not sure what you mean by "tuple of type".
Unless the type defines a compatible `computeHash` function OR can be sliced into a `const(ubyte)[]`, we call `tupleof`, which iterates over every field in sequence.
Note that this is calling `tupleof` on the instance, so it iterates on the field. If we call `tupleof` on the type, we iterate on the types of fields.

Reply:
@AndrejMitrovic : Fixed